added santander data

nishaq503 opened the pull request:
feature: added santander dataset to benchmark suite

thoward27 commented on src/__main__.py:
```diff
@@ -130,9 +142,10 @@ def test(method, dataset, metric, min_points, graph_ratio):
 @click.option('--min-points', type=str, default='*')
 @click.option('--graph-ratio', type=str, default='*')
-def plot_results(plot, method, dataset, metric, min_points, graph_ratio):
+def plot_results(plot, method, dataset, metric, starting_depth, min_points, graph_ratio):
```
maybe starting depth should be depth - starting depth? We won't always have starting_depth layers, will be?

thoward27 commented on src/__main__.py:
```diff
@@ -143,13 +156,13 @@ def plot_results(plot, method, dataset, metric, min_points, graph_ratio):
             manifold = Manifold.load(fp, data)
         for method in methods:
-            for depth in range(0, manifold.depth + 1):
+            for depth in range(starting_depth, manifold.depth + 1):
```
what if starting depth > manifold.depth?

thoward27 commented on src/datasets.py:
```diff
@@ -61,18 +62,20 @@ def get(dataset: str) -> None:
 
     if dataset == 'p53mutants':
         _get_mutants(filename)
+    if dataset == 'santander':
```
should be at least an elif, if not a Dict[str, callable]

thoward27 commented on src/datasets.py:
```diff
@@ -61,18 +62,20 @@ def get(dataset: str) -> None:
     try:
         data_dict = scipy.io.loadmat(filename)
-    except NotImplementedError:
+    except (NotImplementedError, MatReadError):
```
does this catch too much? feels right I just want to make sure

thoward27 commented on src/methods.py:
```diff
@@ -1,9 +1,8 @@
 import numpy as np
-
-from chess.manifold import Manifold, Graph, Cluster
+from pyclam.manifold import Graph, Cluster
```
Weird that we don't need manifold.. oh well

thoward27 reviewed (approved):
There are a few very small details to tweak, then LGTM

Commit:
added santander data

feature: added santander dataset to benchmark suite
added import

fix: added import
improved cli arguments


incorporated new manifold graph building


reordered datasets


finalized defaults for anomaly experiments


updated pyclam


alphabetical order on methods

Commit:
better subsampling


better min-points


properly using random walks


made subsampling into global variable
